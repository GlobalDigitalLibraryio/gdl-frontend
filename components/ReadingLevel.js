// @flow
/**
 * Part of GDL gdl-frontend.
 * Copyright (C) 2017 GDL
 *
 * See LICENSE
 */

import * as React from 'react';
import styled from 'react-emotion';
import { Trans } from '@lingui/react';
import media from '../style/media';
import theme from '../style/theme';

const Label = styled('span')`
  box-shadow: 0 0 2px 0 rgba(0, 0, 0, 0.12), 0 2px 2px 0 rgba(0, 0, 0, 0.12);
  border-radius: 4px;
  background-color: ${theme.colors.grayDarker};
  color: ${theme.colors.white};
  padding-left: 10px;
  padding-right: 10px;
  font-size: 11px;
  line-height: 17px;
  ${media.tablet`
    font-size: 12px;
    line-height: 18px;
  `};
`;

export default ({ level, ...props }: { level: string }) => (
  <Label {...props}>
    <Trans>Level {level}</Trans>
  </Label>
);
