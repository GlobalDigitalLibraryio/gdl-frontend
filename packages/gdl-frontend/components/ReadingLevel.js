// @flow
/**
 * Part of GDL gdl-frontend.
 * Copyright (C) 2017 GDL
 *
 * See LICENSE
 */

import * as React from 'react';
import styled from 'react-emotion';

import type { ReadingLevel } from '../types';
import ReadingLevelTrans from './ReadingLevelTrans';
import media from '../style/media';
import { colors, misc } from '../style/theme';

const Label = styled('span')`
  box-shadow: 0 0 2px 0 rgba(0, 0, 0, 0.12), 0 2px 2px 0 rgba(0, 0, 0, 0.12);
  border-radius: ${misc.radius};
  background-color: ${colors.base.grayDarkest};
  color: ${colors.base.white};
  padding-left: 10px;
  padding-right: 10px;
  font-size: 11px;
  line-height: 17px;
  ${media.tablet`
    font-size: 12px;
    line-height: 18px;
  `};
`;

type Props = {|
  level: ReadingLevel
|};

export default ({ level }: Props) => (
  <div css={{ textAlign: 'center' }}>
    <Label>
      <ReadingLevelTrans readingLevel={level} />
    </Label>
  </div>
);
