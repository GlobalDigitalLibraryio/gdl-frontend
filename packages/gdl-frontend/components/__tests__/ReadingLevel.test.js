// @flow
/**
 * Part of GDL gdl-frontend.
 * Copyright (C) 2017 GDL
 *
 * See LICENSE
 */

import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { createSerializer } from 'jest-emotion';
import * as emotion from 'emotion';
import ReadingLevel from '../ReadingLevel';

expect.addSnapshotSerializer(createSerializer(emotion));

test('Renders', () => {
  const tree = mount(<ReadingLevel level="1" />);

  expect(toJson(tree)).toMatchSnapshot();
});
