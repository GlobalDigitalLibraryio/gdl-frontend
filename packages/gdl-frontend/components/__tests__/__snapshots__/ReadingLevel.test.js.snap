// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders 1`] = `
.emotion-0 {
  box-shadow: 0 0 2px 0 rgba(0,0,0,0.12),0 2px 2px 0 rgba(0,0,0,0.12);
  border-radius: 4px;
  background-color: #666;
  color: #fff;
  padding-left: 10px;
  padding-right: 10px;
  font-size: 11px;
  line-height: 17px;
}

@media (min-width:768px) {
  .emotion-0 {
    font-size: 12px;
    line-height: 18px;
  }
}

<Component
  level="1"
>
  <Styled(span)>
    <span
      className="emotion-0 emotion-1"
    >
      <ReadingLevelTrans
        readingLevel="1"
      >
        <withI18n
          id="Level {level}"
          values={
            Object {
              "level": "1",
            }
          }
        >
          <Trans
            id="Level {level}"
            values={
              Object {
                "level": "1",
              }
            }
          >
            <Render
              value="Level {level}"
            >
              Level {level}
            </Render>
          </Trans>
        </withI18n>
      </ReadingLevelTrans>
    </span>
  </Styled(span)>
</Component>
`;
